codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerBuilderGeneratorTest.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy.generators
+
+import org.junit.jupiter.api.Test
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.rust.codegen.core.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.core.rustlang.rust
+import software.amazon.smithy.rust.codegen.core.smithy.generators.StructureGenerator
+import software.amazon.smithy.rust.codegen.core.smithy.generators.implBlock
+import software.amazon.smithy.rust.codegen.core.testutil.asSmithyModel
+import software.amazon.smithy.rust.codegen.core.testutil.compileAndTest
+import software.amazon.smithy.rust.codegen.core.testutil.generatePluginContext
+import software.amazon.smithy.rust.codegen.core.util.lookup
+import software.amazon.smithy.rust.codegen.server.smithy.renderDefaultBuilder
+import software.amazon.smithy.rust.codegen.server.smithy.testutil.serverTestCodegenContext
+import software.amazon.smithy.rust.codegen.server.smithy.testutil.serverTestSymbolProvider
+import kotlin.io.path.createDirectory
+import kotlin.io.path.writeText
+
+class ServerBuilderGeneratorTest {
+    @Test
+    fun `generate default values`() {
+        val model =
+            """
+            namespace com.test
+            use aws.protocols#restJson1
+            use smithy.framework#ValidationException
+
+            structure MyStruct {
+               @required
+               requiredFoo: String = "required foo",
+               foo: String = "foo",
+               bar: PrimitiveInteger = 42,
+               baz: Integer = 42,
+               baw: Float = 42.0,
+               yes: Boolean = true,
+               no: Boolean = false,
+            }
+
+            @default(42)
+            integer PrimitiveInteger
+            """.asSmithyModel(smithyVersion = "2")
+        val provider = serverTestSymbolProvider(model)
+        val writer = RustWriter.forModule("model")
+        writer.rust("##![allow(deprecated)]")
+        val struct = model.lookup<StructureShape>("com.test#MyStruct")
+
+        val (_, testDir) = generatePluginContext(model)
```

Comment:
This test is using the old testing machinery. Use the modern way of writing unit tests (i.e. starting from a project `val project = TestWorkspace.testProject(symbolProvider)`.